plugins/frontend/backstage-plugin-cloudsmith/src/api/CloudsmithClient.ts

```diff
@@ -53,4 +67,58 @@ export class CloudsmithClient implements CloudsmithApi {
     }
     return await response.json();
   }
+
+  // get data from Cloudsmith quota endpoint (https://help.cloudsmith.io/reference/quota_read)
+  async getQuota({ owner }: { owner: string }): Promise<CloudsmithUsage> {
+    const cloudsmithApiUrl = await this.getApiUrl();
+    const response = await this.fetchApi.fetch(
+      `${cloudsmithApiUrl}/quota/${owner}/`,
+    );
+    if (!response.ok) {
+      throw new Error(`Failed to retrieve quota: ${response.statusText}`);
+    } else {
+      if (response.status === 402) {
+        throw new Error(`Payment Required: ${response.statusText}`);
+      }
+    }
+    return await response.json();
+  }
+
+  // get repository audit logs from Cloudsmith audit repo logs endpoint (https://help.cloudsmith.io/reference/audit_log_repo_list) and support pagniation
+  async getRepoAuditLogs({
+    owner,
+    repo,
+  }: {
+    owner: string;
+    repo: string;
+  }): Promise<RepoAuditLog> {
+    const cloudsmithApiUrl = await this.getApiUrl();
+    const response = await this.fetchApi.fetch(
+      `${cloudsmithApiUrl}/audit-log/${owner}/${repo}/?page_size=100`,
+    );
+    if (!response.ok) {
+      throw new Error(`Failed to retrieve audit logs: ${response.statusText}`);
+    }
+    return await response.json();
+  }
+
+  // get repository security scan logs for a Cloudsmith repository endpoint (https://help.cloudsmith.io/reference/vulnerabilities_repo_list)
+  async getRepoSecurityScanLogs({
+    owner,
+    repo,
+  }: {
+    owner: string;
+    repo: string;
+  }): Promise<RepoVulnerability> {
+    const cloudsmithApiUrl = await this.getApiUrl();
+    const response = await this.fetchApi.fetch(
+      `${cloudsmithApiUrl}/vulnerabilities/${owner}/${repo}/?page_size=100`,
+    );
+    if (!response.ok) {
```

Comment:
I think it would be nice to do a check for the paywall and handle it in the components. Then you can send users to the cloudsmith marketting site and hopefully make a sale?

Reply:
Maybe later if you prefer.